authorization/invitation/service/invitation_service.go

```diff
@@ -381,76 +382,95 @@ func (s *invitationServiceImpl) Accept(ctx context.Context, token uuid.UUID) (st
 		return "", redirectOnFailure, autherrors.NewUnauthorizedError("user deprovisioned")
 	}
 
-	invitationID := inv.InvitationID
-	// If this invitation is for an identity
-	if inv.InviteTo != nil {
-		inviteTo := *inv.InviteTo
-		inviteToIdentity, err := s.Repositories().Identities().Load(ctx, inviteTo)
-		if err != nil {
-			return "", redirectOnFailure, errs.Wrapf(err, "failed to load identity for inviteTo %d", inviteTo)
-		}
+	var resourceID string
 
-		// If the invitation is for a membership, add a membership record
-		if inv.Member {
-			s.Repositories().Identities().AddMember(ctx, inviteToIdentity.ID, currentIdentityID)
-		}
+	err = s.ExecuteInTransaction(func() error {
 
-		roles, err := s.Repositories().InvitationRepository().ListRoles(ctx, invitationID)
-		if err != nil {
-			return "", redirectOnFailure, errs.Wrapf(err, "failed to list roles for invitation %d", invitationID)
-		}
+		// If this invitation is for an identity
+		if inv.InviteTo != nil {
+			inviteToIdentity, err := s.Repositories().Identities().Load(ctx, *inv.InviteTo)
+			if err != nil {
+				return err
+			}
 
-		// If the invitation includes role assignments, assign them
-		inviteToIdentityResourceID := inviteToIdentity.IdentityResourceID.String
-		for _, role := range roles {
-			ir := &repository.IdentityRole{
-				IdentityID: currentIdentityID,
-				RoleID:     role.RoleID,
-				ResourceID: inviteToIdentityResourceID,
+			// If the invitation is for a membership, add a membership record
+			if inv.Member {
+				err = s.Repositories().Identities().AddMember(ctx, inviteToIdentity.ID, currentIdentityID)
+				if err != nil {
+					return err
+				}
 			}
 
-			err = s.Repositories().IdentityRoleRepository().Create(ctx, ir)
+			roles, err := s.Repositories().InvitationRepository().ListRoles(ctx, inv.InvitationID)
 			if err != nil {
-				return "", redirectOnFailure, errs.Wrap(err, "failed to create identity role")
+				return err
```

Comment:
also, the error could be wrapped with a message to given some context information of where/when the `ListRoles` method was called

Reply:
See previous response re. error handling.